packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -460,7 +490,8 @@ export class TreeCheckout implements ITreeCheckoutFork {
 								}
 								this.revertRevertible(revision, data.kind);
 								if (release) {
-									revertible.dispose();
+									this.disposeRevertible(revertible, revision);
+									onRevertibleDisposed?.(revertible);
```

Comment:
This was to fix a bug with the manual dispose which causes the output of reverts to be disposed right away since the data is disposing everything with the same major (which I think you pointed out in another comment somewhere). But yeah, not necessary at all since we can remove disposeRevertibleRoots

Reply:
I did not understand the explanation but I'm resolving the thread for now. We can revisit if need be.